packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -322,10 +326,37 @@ async function findAndInvalidateLinks(
       return;
     }
 
+    const patient = await getPatientOrFail({ cxId, id: patientId });
+    const cwAccess = await getCWAccessForPatient(patient);
+    if (cwAccess.error != null) throw new MetriportError(cwAccess.error);
+    const { commonWell, queryMeta } = cwAccess;
+
+    const downgradeRequests: Promise<NetworkLink>[] = [];
+    for (const link of invalidLinks.commonwell) {
+      if (!link._links?.downgrade?.href) continue;
+      const downgradeRequest = commonWell
+        .upgradeOrDowngradeNetworkLink(queryMeta, link._links.downgrade.href)
+        .catch(error => {
+          const msg = `Failed to downgrade link`;
+          log(`${msg}. Cause: ${errorToString(error)}`);
+          capture.error(msg, {
+            extra: {
+              patientId,
+              cxId,
+              cwReference: commonWell.lastReferenceHeader,
+              error,
+            },
+          });
+          throw error;
+        });
+      downgradeRequests.push(downgradeRequest);
+    }
+
     await Promise.all([
       createOrUpdateInvalidLinks({ id: patientId, cxId, invalidLinks }),
       updateCQPatientData({ id: patientId, cxId, cqLinksToInvalidate: invalidLinks.carequality }),
       updateCwPatientData({ id: patientId, cxId, cwLinksToInvalidate: invalidLinks.commonwell }),
+      ...downgradeRequests,
```

Comment:
Prob fine on this context, since we should have so many links, but important that we avoid `Promise.all` on a list of promises in most cases.